packages/cozy-harvest-lib/src/services/biUtils.js

```diff
@@ -0,0 +1,102 @@
+import assert from '../assert'
+import biPublicKeyProd from './bi-public-key-prod.json'
+
+const biURLProd = 'https://cozy.biapi.pro/2.0'
+const biURLDev = 'https://cozytest-sandbox.biapi.pro/2.0'
+
+export const getBIModeFromCurrentLocation = (wdw = window) => {
+  if (!wdw || !wdw.location || !wdw.location.host) {
+    return 'dev'
+  } else {
+    const domain = wdw.location.host
+      .split('.')
+      .slice(-2)
+      .join('.')
+    switch (domain) {
+      case 'cozy.rocks':
+      case 'mycozy.cloud':
+        return 'prod'
+      case 'cozy.works':
+      case 'cozy.dev':
+        return 'dev'
+      default:
+        return 'dev'
+    }
+  }
+}
+
+export const getBIConfig = mode => {
+  if (mode === 'prod') {
+    return {
+      url: biURLProd,
+      publicKey: biPublicKeyProd
+    }
+  } else {
+    return {
+      url: biURLDev
+    }
+  }
+}
+
+const biRequest = async (method, path, config, rawForm, bearer) => {
```

Comment:
Not a big deal, but I'm surprised this isn't in a separate package. Do we not need these functions elsewhere?

Reply:
I extracted only a small part of the bi API from the banking konnectors. I only mutualized the encryption and translation of credential fields since otherwise we would have needed to copy/paste all the field definitions and this is subject to change. The bi request part is very stable and the code right now in the banking connectors is not suitable for being pruned (see https://gitlab.cozycloud.cc/labs/banking-konnectors/blob/master/src/bi/bi-api.js for all the supported APIs inside the banking konnector), this is why I recoded it here.

I agree with the premise here though, that a bi-sdk package could be of use. I a going to leave a comment.